### PR TITLE
Fix elm module resolve

### PR DESCRIPTION
**Pull request by mblarsen**
When importing Elm modules from a non root dir they would not be imported correctly. Expected wrong names.

With this the default is to have Elm source files in ./src (same as entry file)

You can override this with loaderOption cwd, e.g.:

```javascript
presets: [
    require('./index.js')({
      loaderOptions: {
        cwd: __dirname + '/example',
      },
    })
  ]
```

I considered instead taking the dir of the last entry file and use instead (in devserver mode the first entry unshifted by poi), but opted to leave it manual.

### CODE REVIEW

File: packages/poi-preset-elm/index.js [hunk 1]
```diff
@@ -1,13 +1,16 @@
 const fs = require('fs')
+const path = require('path')
 
 module.exports = (options = {}) => {
   return poi => {
     const localCompiler = 'node_modules/.bin/elm-make'
     const loaderOptions = Object.assign({
       // Only use local compiler when it exists
-      pathToMake: fs.existsSync(localCompiler) ? localCompiler : null,
+      pathToMake: fs.existsSync(localCompiler) ? path.join(__dirname, localCompiler) : null,
```
Comment (egoist) [inline packages/poi-preset-elm/index.js]: why `__dirname`?

File: packages/poi-preset-elm/index.js [hunk 2]
```diff
@@ -1,13 +1,16 @@
...
       warn: true,
-      debug: poi.options.mode !== 'production'
+      debug: poi.options.mode !== 'production',
+      cache: false,
+      cwd: path.join(__dirname, '/src')
```
Comment (egoist) [inline packages/poi-preset-elm/index.js]: maybe should be `process.cwd()` instead of `__dirname` by default?

**Comment by egoist** [comment]
hmm I'm not sure about how this works, here it says https://github.com/elm-community/elm-webpack-loader#cwd-default-null-recommended `cwd` is recommended to be the same directory as your `elm-package.json` file

File: packages/poi-preset-elm/index.js (hunk 2 above)
Comment (mblarsen) [reply packages/poi-preset-elm/index.js]: Agreed

File: packages/poi-preset-elm/index.js (hunk 1 above)
Comment (mblarsen) [reply packages/poi-preset-elm/index.js]: localCompiler value becomes relative to what `cwd` is. So if changed to something else the elm-make bin path is no longer valid. 

Should probably be `process.cwd()` as you suggest, but should not be left out.

**Comment by mblarsen** [comment]
Yes, I know of the recommendation of `cwd` however with poi's other default of the entry file `src/index.js` there is a bit of a clash. 

If `cwd` is set to null then modules that are in the same dir as Main.elm cannot be found.

```
Error in ~/workspace/egoist/poi/packages/poi-preset-elm/example/index.js

Error: Compiler process exited with error Compilation failed
I cannot find module 'Const'.

Module 'Main' is trying to import it.
```

This is because Elm mirrors the module names in the file structure. So for it to work we must change:

```
import Html exposing (..)
import Html.Events exposing (onClick)
import Const exposing (startValue, stepValue)
```

to

```
import Html exposing (..)
import Html.Events exposing (onClick)
import Example.Const exposing (startValue, stepValue)
```
(change should also happen in module itself)

even though `Const.elm` is in the same dir as `Main.elm`. Also, note that the module names are title case (not sure this would work in a case-sensitive OS).

So you get this awkward module names because of poi's defaults.

I'll play around with it a bit to see if I can come up with a better solution.

**Comment by mblarsen** [comment]
> cwd is recommended to be the same directory as your elm-package.json file

It creates the file and elm-stuff wherever cwd is. So if I were to set cwd to example those files would be created in that dir.

**Comment by mblarsen** [comment]
I'm starting to think that the problem is a non-issue when I see this error trace:

```
 (/Users/mblarsen/workspace/egoist/poi/node_modules/node-elm-compiler/index.js:141:27)
```

When running it "drops back" to use `egoist/poi/node_modules` and not my setup for the cwd. I tried it out in a separate setup and works.

I'll clean up and commit some minimal improvements.

**Comment by mblarsen** [comment]
Going in circles. I will look at this later 😭

**Pull request closed without merging**